crates/interpreter/src/instructions/opcode.rs

```diff
@@ -42,6 +42,24 @@ impl<H: Host> InstructionTables<'_, H> {
 }
 
 impl<'a, H: Host + 'a> InstructionTables<'a, H> {
+    /// Inserts a boxed instruction into the table with the specified index.
+    ///
+    /// Returns an error if the table is not boxed.
+    #[inline]
+    pub fn insert_boxed(
+        &mut self,
+        opcode: u8,
+        instruction: BoxedInstruction<'a, H>,
+    ) -> Result<(), InsertError> {
+        match self {
+            Self::Plain(_) => Err(InsertError::NotBoxed),
```

Comment:
should we convert to boxed instead?

Reply:
I would panic here. Silently converting can have a side-effect on performance

Reply:
this would make evm setup fallible and force unwrap with a preceding call to to_boxed, i think this makes it a lot less convenient to use, and since additional instructions are only supported by boxing all of them, converting them here seems reasonable to me

Reply:
That is the point to fail and make it explicit to call, this would look something like.
```
table.convert_to_box();
table.insert_boxed(instr);
```
It is redundant but explicit on what is going on.

Not the point I would die on, so lets convert it to boxed inside `insert_boxed`.

Reply:
Changed to convert into the boxed variant inside `insert_boxed`. updated docs and the test which no longer needs a map